### PR TITLE
Add monitoring for Flux

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/19449 and https://github.com/giantswarm/giantswarm/issues/19455

### CODE REVIEW

File: hack/additional-resources/metrics-service.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+spec:
+  selector:
+    app.kubernetes.io/name: '{{ include "name" . | quote }}'
+    app.kubernetes.io/instance: "{{ .Release.Name | quote }}"
+    app.kubernetes.io/managed-by: "{{ .Release.Service | quote }}"
+  clusterIP: "None"
+  ports:
+    - protocol: TCP
+      port: 8080
+      targetPort: http-prom
```
Comment (MarcelMue) [inline hack/additional-resources/metrics-service.yaml:20]: Should we template this? So `http-prom` is not hardcoded.

**Review by MarcelMue — APPROVED** [review]
Generally ok with me if tested - just a minor comment

**Comment by kubasobon** [comment]
Trying to fix 
```yaml
  reason: 'Upgrade "flux-app" failed: failed to create resource: PodSecurityPolicy.extensions
    "\"flux-app-pvc-psp\"" is invalid: metadata.name: Invalid value: "\"flux-app-pvc-psp\"":
    a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters,
    ''-'' or ''.'', and must start and end with an alphanumeric character (e.g. ''example.com'',
    regex used for validation is ''[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'')'
```

**Comment by kubasobon** [comment]
Success!

**Pull request merged**